src/vs/workbench/services/themes/electron-browser/workbenchThemeService.ts

```diff
@@ -514,21 +527,61 @@ export class WorkbenchThemeService implements IWorkbenchThemeService {
 				return true;
 			}
 		}
+
+		if (!objects.equals(newTokenColorCustomizations, this.tokenColorCustomizations)) {
+			return true;
+		}
+
 		return false;
 	}
 
+	private getTokenColorCustomizations(): ITokenColorizationRule[] {
+		let tokenColorCustomizations = this.configurationService.lookup<ITokenColorCustomization>(CUSTOM_EDITOR_COLORS_SETTING).value || {};
+		let rules: ITokenColorizationRule[] = [];
+
+		let value, settings, scope;
+		Object.keys(tokenGroupToScopesMap).forEach(key => {
+			value = tokenColorCustomizations[key];
+			settings = typeof value === 'string'
+				? { foreground: value }
+				: value;
+			scope = tokenGroupToScopesMap[key];
+
+			if (!settings) {
+				return;
+			}
+
+			rules.push({
+				scope,
+				settings
+			});
+		});
+
+		// Put the general customizations such as comments, strings, etc. first so that
+		// they can be overriden by specific customizations like "string.interpolated"
+		const textMateRules: ITokenColorizationRule[] = tokenColorCustomizations['textMateRules'] || [];
+
+		return rules.concat(textMateRules);
+	}
+
 	private updateColorCustomizations(notify = true): void {
-		let newColorCustomizations = this.configurationService.lookup<IColorCustomizations>(CUSTOM_COLORS_SETTING).value || {};
+		let newColorCustomizations = this.configurationService.lookup<IColorCustomizations>(CUSTOM_WORKBENCH_COLORS_SETTING).value || {};
 		let newColorIds = Object.keys(newColorCustomizations);
 		if (newColorIds.length === 0) {
 			newColorCustomizations = this.configurationService.lookup<IColorCustomizations>(DEPRECATED_CUSTOM_COLORS_SETTING).value || {};
 			newColorIds = Object.keys(newColorCustomizations);
 		}
-		if (this.hasCustomizationChanged(newColorCustomizations, newColorIds)) {
+
+		let newRules = this.getTokenColorCustomizations();
```

Comment:
Setting change notifications can happen quite often, cause by any change to settings. Mostly the change does not affect the custom colors. For that I tried to optimize the test for changes and avoid the creation of new data structures unless there is a change to the colors.
I'd suggest to not construct the rules array as in getTokenColorCustomizations, but just compare the old to the new settings.
The construction of the rules (as done in getTokenColorCustomizations) could be done inside setCustomTokenColors, like it is done with custom colors.